### PR TITLE
allow setting validate certificate flag from config

### PR DESCRIPTION
**Pull request by ryuyu**
@skofman1 @scottbommarito @maartenba

### CODE REVIEW

**Comment by scottbommarito** [comment]
LGTM

File: NuGet.Services.Dashboard.Common/SecretReaderFactory.cs [hunk 1]
```diff
@@ -11,12 +11,16 @@ public class SecretReaderFactory: ISecretReaderFactory
...
+        private bool _validateCertificate;
 
         public SecretReaderFactory(NameValueCollection config)
         {
             _vaultName = config["keyVault-VaultName"];
             _clientId = config["keyVault-ClientId"];
             _certificateThumbprint = config["keyVault-CertificateThumbprint"];
+            _validateCertificate = bool.TryParse(config["keyVault-ValidateCertificate"], out _validateCertificate)
+                                    ? _validateCertificate
+                                    : true;
```
Comment (skofman1) [inline NuGet.Services.Dashboard.Common/SecretReaderFactory.cs]: if u can't parse u should fail the flow

File: NuGet.Services.Dashboard.FrontEnd/Web.config [hunk 2]
```diff
@@ -31,6 +31,7 @@
     <add key="keyVault-VaultName" value="" />
     <add key="keyVault-ClientId" value="" />
     <add key="keyVault-CertificateThumbprint" value="" />
+    <add key="keyVault-ValidateCertificate" value=""/>
```
Comment (skofman1) [inline NuGet.Services.Dashboard.FrontEnd/Web.config]: set to false by default

**Comment by maartenba** [comment]
Any reason why false is default?

**Comment by skofman1** [comment]
Only for this service.. we didn't find a way to make the cert trusted on the WebApp. We can set the default to true and override in the config. Do you prefer that?

**Comment by maartenba** [comment]
Yeah I'd like us to have the defaults to be restrictive for this type of thing, and make it a conscious override

**Pull request merged**